internal/catalogmetadata/filter/bundle_predicates.go

```diff
@@ -31,6 +31,19 @@ func InMastermindsSemverRange(semverRange *mmsemver.Constraints) Predicate[catal
 	}
 }
 
+func HigherBundleVersion(currentVersion *bsemver.Version) Predicate[catalogmetadata.Bundle] {
+	return func(bundle *catalogmetadata.Bundle) bool {
+		if currentVersion == nil {
+			return false
+		}
+		bundleVersion, err := bundle.Version()
```

Comment:
Not for this PR, just a though. I wonder if we should validate the bundle version on creation/ingress rather than on egress (i.e. bundle.Version() shouldn't return an error). So, a bundle object is always validated and correct.

Reply:
I've had this thought too. If it didn't break any assumptions, it would be nice to get something into the `declcfg.Bundle` type for this.

Reply:
Yeah that makes great sense and also would make it a lot easier to grab.

Reply:
I'll create a discussion upstream - see what ppl think - or would a ticket suffice?